webapp/src/Controller/API/UserController.php

```diff
@@ -64,27 +64,80 @@ public function __construct(
      *     name="tsv",
      *     in="formData",
      *     type="file",
-     *     required=true,
+     *     required=false,
      *     description="The groups.tsv files to import."
      * )
+     * @SWG\Parameter(
+     *     name="json",
+     *     in="formData",
+     *     type="file",
+     *     required=false,
+     *     description="The groups.json files to import."
+     * )
      * @SWG\Response(
      *     response="200",
      *     description="Returns a (currently meaningless) status message.",
      * )
-     * @throws BadRequestHttpException
+     * @throws Exception
      */
-    public function addGroupAction(Request $request)
+    public function addGroupsAction(Request $request)
     {
         /** @var UploadedFile $tsvFile */
         $tsvFile = $request->files->get('tsv') ?: [];
-        if ($this->importExportService->importTsv('groups', $tsvFile, $message)) {
+        /** @var UploadedFile $jsonFile */
+        $jsonFile = $request->files->get('json') ?: [];
+        if (!$tsvFile && !$jsonFile) {
+            throw new BadRequestHttpException('Neither TSV nor JSON file supplied');
```

Comment:
nit: merge the two exceptions into one which you phrase like this:
`Supply exactly one of 'json' or 'tsv'.`